src/HCSR04.cpp

```diff
@@ -20,21 +22,30 @@ double UltraSonicDistanceSensor::measureDistanceCm() {
 }
 
 double UltraSonicDistanceSensor::measureDistanceCm(float temperature) {
+    uint32_t maxDistanceDurationMicroSec;
+
     // Make sure that trigger pin is LOW.
     digitalWrite(triggerPin, LOW);
     delayMicroseconds(2);
     // Hold trigger for 10 microseconds, which is signal for sensor to measure distance.
     digitalWrite(triggerPin, HIGH);
     delayMicroseconds(10);
     digitalWrite(triggerPin, LOW);
+    double speedOfSoundInCmPerMicroSec = 0.03313 + 0.0000606 * temperature; // Cair ≈ (331.3 + 0.606 ⋅ ϑ) m/s
+
+    // Compute max delay based on max distance with 25% margin in microseconds or use absolute timeout
+    maxDistanceDurationMicroSec = 2.5 * maxDistanceCm / speedOfSoundInCmPerMicroSec;
+    if (timeoutMicroSec > 0) {
+    	min(maxDistanceDurationMicroSec, timeoutMicroSec);
```

Comment:
Hm well this doesn't do anything, expression is not assigned!

How about we do
```
uint32_t maxDistanceDurationMicroSec = 2.5 * maxDistanceCm / speedOfSoundInCmPerMicroSec;
uint32_t timeoutMicroSec = min(maxDistanceDurationMicroSec, maxTimeoutMicroSec);

unsigned long durationMicroSec = pulseIn(echoPin, HIGH, timeoutMicroSec);

```

Notice I moved the declaration of maxDistanceDurationMicroSec from above diretly to the defintion -> we arein cpp so we can do this.

Reply:
you're right, fixed